treeherder/perf/management/commands/reassign_perf_data.py

```diff
@@ -0,0 +1,50 @@
+from django.core.management.base import (BaseCommand,
+                                         CommandError)
+from django.db import transaction
+
+from treeherder.perf.models import PerformanceAlert, PerformanceDatum
+
+
+class Command(BaseCommand):
+    help = """
+    Moves performance data from an old signature to
+    a new one.
+    Useful when renaming perf tests, as it allows
+    to combine old signature data with new one.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--from',
+            action='append',
+            help='Original signature (specify multiple times to get multiple signatures)'
+        )
+        parser.add_argument(
+            '--to',
+            action='append',
+            help='New signature we want to move performance data to '
+                 '(specify multiple times to get multiple signature)'
+        )
+
+    def handle(self, *args, **options):
+        from_signatures = options['from']
+        to_signatures = options['to']
+
+        if not (from_signatures and to_signatures):
+            raise CommandError("Must specify both old and new signatures")
+        if len(from_signatures) != len(to_signatures):
+            raise CommandError("Each old signature must have a corresponding new one")
+
+        for from_sign, to_sign in zip(from_signatures, to_signatures):
```

Comment:
Feels like some sanity check would help, prior to doing the reassignment.
I'd say checking that both signatures from each pair have the same repository & framework & platform & option_collection & extra_options would prevent from bad reassignment.
After all, we're only interested in updating test names and maybe the suite as well.

Reply:
@sarah-clements is it ok if I ask for your feedback on this?

Reply:
I may be missing some information about the effects of renaming, but wouldn't it be easier to reduce redundant subtest information dynamically in a serializer (which would require changing the viewsets) like what was done  [here](https://github.com/mozilla/treeherder/blob/master/treeherder/webapp/api/performance_serializers.py#L194) and [here](https://github.com/mozilla/treeherder/blob/master/treeherder/webapp/api/serializers.py#L272)?  It'd be worth having @edmorley's feedback as well.

Reply:
I just remembered that the names for tests and subtests in Perfherder is generated dynamically in the UI for both the series data and performance datum endpoints, [here](https://github.com/mozilla/treeherder/blob/master/ui/models/perfSeries.js#L55). (When I created the new perf API, I used some of the naming logic from there but put it into the serializer mentioned in the previous comment).

Reply:
> 
> 
> I may be missing some information about the effects of renaming, but wouldn't it be easier to reduce redundant subtest information dynamically in a [serializer](https://www.django-rest-framework.org/api-guide/serializers/#serializers) (which would require changing the corresponding API's use of viewsets) like what was done [here](https://github.com/mozilla/treeherder/blob/master/treeherder/webapp/api/performance_serializers.py#L194)? It'd be worth having @edmorley's feedback as well.

That's a good option for shortening names, of course.  Thing is we have another use case in which we need to rename test/suite names so they are more precise and self describing (warm vs cold test runs).